### PR TITLE
Misx fixes and improvements to Azure metric export

### PR DESCRIPTION
**Pull request by asheniam**
Hi Conor,

I stumbled across your Azure Metric exporter for Prometheus and started experimenting around with it.  I have few suggestions / updates:
a. Fixes for encoding Azure metric name to correctly match Prometheus metric name conventions
b. Query Azure Metric API with latest timestamp and use query delay of 3 minutes
c. Export Azure metric average/total/min/max aggregation types into Prometheus

I verified the changes e2e locally and I'm able to pull VM metrics from my Azure subscription into Prometheus.

Cheers
- Andy

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -51,14 +51,36 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
 				log.Printf("No metric data returned for metric %v at target %v\n", metric.Name, target.Resource)
 				continue
 			}
-			metricName := metricValueData.Value[0].Name.Value
+
+			// Ensure Azure metric names conform to Prometheus metric name conventions
```
Comment (Conorbro) [inline main.go]: We had a think about this and decided it would be best to leave the metric names as they were from source.

File: utils.go [hunk 2]
```diff
@@ -17,12 +17,15 @@ func PrintPrettyJSON(input map[string]interface{}) {
 	fmt.Println(string(out))
 }
 
-// GetTimes - Returns the current time and the time 1 minute ago.
+// GetTimes - Returns the endTime and startTime used for querying Azure Metrics API
```
Comment (Conorbro) [inline utils.go]: Can you explain the reasoning for this change?

File: main.go [hunk 3]
```diff
@@ -51,14 +51,36 @@ func (c *Collector) Collect(ch chan<- prometheus.Metric) {
...
 			labels := CreateResourceLabels(metricValueData.Value[0].ID)
+
 			ch <- prometheus.MustNewConstMetric(
-				prometheus.NewDesc(metricName, "", nil, labels),
+				prometheus.NewDesc(metricName+"_total", "", nil, labels),
 				prometheus.GaugeValue,
 				metricValue.Total,
 			)
+
+			ch <- prometheus.MustNewConstMetric(
```
Comment (Conorbro) [inline main.go]: Can you clarify what the average is in this case and what the min and max are (over what time period essentially)?

File: main.go (hunk 1 above)
Comment (asheniam) [reply main.go]: How do you plan to handle cases where Prometheus metric names don't allow specific characters?  For example, "Percentage CPU" for VM metrics.  I discovered this issue when trying to run this project to export Azure VM metrics into Prometheus.

File: main.go (hunk 3 above)
Comment (asheniam) [reply main.go]: The Azure Monitor Metrics REST API returns back 1 minute metrics, essentially aggregations over 1 minute time period.  The total/average/min/max represent the corresponding aggregations for that 1 minute timestamp.  For example, if the metric is Latency -- then avg = average latency in that 1 minute, min = minimum latency in that 1 minute, etc.  Depending on the metric, there might be different aggregation types which make more sense for that specific metric.  There is also a concept of the PrimaryAggregationType for each metric which can be discovered by calling the Metric Definitions REST API.  This is the aggregation type that the Azure service team decided was the most relevant aggregation (if only 1 aggregation type was to be plotted).

File: utils.go (hunk 2 above)
Comment (asheniam) [reply utils.go]: My understanding of how this works is that Prometheus will periodically poll / scrape the results from this project's endpoint.   This project should be querying for the latest metric data from Azure Monitor Metrics API -- there is typically ~ 3 minute lag in the metric data so I used this as the query delay.  Also we should be querying the Azure Monitor Metrics API with timestamps in UTC.

**Comment by Conorbro** [comment]
Thanks!

**Pull request merged**